test/integration/test-amp-recaptcha-input.js

```diff
@@ -0,0 +1,255 @@
+            <p>recaptcha-token: {{recaptcha-token}}</p>
+          </template>
+        </div>
+
+        <div submit-error>
+          <template type="amp-mustache">
+            <div id="submit-error"></div>
+            <h1>Error! Please check the JS Console in your dev tools.</h1>
+            <p>{{error}}</p>
+          </template>
+        </div>
+
+      </form>
+    </div>
+    `,
+  css: `
+      form.amp-form-submit-success [submit-success] {
+        color: green;
+      }
+      form.amp-form-submit-error [submit-error] {
+        color: red;
+      }
+      form.amp-form-submit-success.hide-inputs > input {
+        display: none;
+      }
+
+      @keyframes donut-spin {
+        0% {
+          transform: rotate(0deg);
+        }
+        100% {
+          transform: rotate(360deg);
+        }
+      }
+
+      .donut {
+        display: inline-block;
+        border: 4px solid rgba(0, 0, 0, 0.1);
+        border-left-color: #7983ff;
+        border-radius: 50%;
+        width: 30px;
+        height: 30px;
+        animation: donut-spin 1.2s linear infinite;
+      }
+
+      .loading-spinner {
+        display: none;
+        text-align: center;
+        margin: 5px;
+      }
+
+      form.amp-form-submitting > .loading-spinner {
+        display: block;
+      }
+    `,
+  /* eslint-enable max-len */
+  extensions: ['amp-recaptcha-input', 'amp-form', 'amp-mustache'],
+  experiments: ['amp-recaptcha-input'],
+}, env => {
+  let doc;
+
+  beforeEach(() => {
+    doc = env.win.document;
+
+    const browserController = new BrowserController(env.win);
+    return browserController.waitForElementLayout('amp-recaptcha-input');
+  });
+
+  it('should be able to create the bootstrap frame', function() {
+    return waitForBootstrapFrameToBeCreated(doc).then(frame => {
+      expect(frame.src.includes('recaptcha')).to.be.true;
+      expect(frame.getAttribute('data-amp-3p-sentinel'))
+          .to.be.equal('amp-recaptcha');
+      expect(frame.getAttribute('name')).to.be.equal(JSON.stringify({
+        'sitekey': '6LebBGoUAAAAAHbj1oeZMBU_rze_CutlbyzpH8VE',
+        'sentinel': 'amp-recaptcha',
+      }));
+    });
+  });
+
+  it('should load the 3p recaptcha frame', function() {
+    return waitForBootstrapFrameOnLoad(doc).then(frame => {
+      expect(frame).to.be.ok;
+    });
+  });
+
+  it('should create a hidden input, ' +
+    'with the value resolved from the recaptcha mock, ' +
+    ' on submit', function() {
+    return submitForm(doc).then(hiddenInput => {
+      expect(hiddenInput).to.be.ok;
+      expect(hiddenInput.name).to.be.equal('recaptcha-token');
+      expect(hiddenInput.value).to.be.equal('recaptcha-mock');
+    });
+  });
+
+  it('should show submit-success on successful submit/response', function() {
+    return submitForm(doc).then(() => {
+      return poll('submit-success', () => {
+        return doc.querySelector('div[id="submit-success"]');
+      }, undefined, 5000);
+    });
+  });
+});
```

Comment:
once you figured out how to do POST, pls add a request-bank test here

Reply:
Done